scantool/diag_l0_elm.c

```diff
@@ -834,6 +910,35 @@ elm_send(struct diag_l0_device *dl0d,
 		fprintf(stderr, FLFMT "ELM: sending %d bytes\n", FL, (int) len);
 	}
 
+	if ((dev->protocol & DIAG_L1_ISO9141) && memcmp(dev->atsh, data, 3)) {
```

Comment:
This is what I was referring to earlier,  regarding atsh[3]; on the first execution this is fragile because atsh is not initialized.